.kokoro/deps-finder.sh

```diff
@@ -28,175 +28,187 @@ scriptDir=$(realpath $(dirname "${BASH_SOURCE[0]}"))
 # include common functions
 source ${scriptDir}/common.sh
 
-#Returns 1 if we find an updated dependency in the commit message.
-function getDependency() {
-  currCommitNumber=$(git rev-parse HEAD)
-
-  #Note that this grep serves to check if we're updating a dependency including shared-dependencies.
-  currCommitMessage=$(git log -1 $currCommitNumber | grep "deps: update dependency com.google.cloud:google-cloud-")
-
-  if [[ -z $currCommitMessage ]]
-  then
-    echo "Commit message does not have correct formatting. Assuming no error."
-    return 1
-  fi
-
-  dependency=${currCommitMessage:28}
-  dependency=${dependency/ to v/:}
-  dependency=${dependency/*com.google.cloud:}
-
-  if [[ -z $dependency ]]
-  then
-    return 1
-  fi
-
-  echo "$dependency"
-  return 0
+#Returns 1 if we find an updated client library in the commit message.
+#Echos client library (if found) before completion.
+function getLibraryUpdate() {
+ currCommitNumber=$(git rev-parse HEAD)
+
+ #Note that this grep serves to check if we're updating a dependency including shared-dependencies.
+ currCommitMessage=$(git log -1 $currCommitNumber | grep "deps: update dependency com.google.cloud:google-cloud-")
+
+ if [[ -z $currCommitMessage ]]
+ then
+   echo "Commit message does not have correct formatting. Assuming no error."
+   return 1
+ fi
+
+ #Use regex to grab the dependency group and artifact IDs
+
+ #Find group and artifact by format
+ groupAndArtifact=$(echo $currCommitMessage | grep -oP 'com.google.cloud:google-cloud-\S+')
+
+ #Grab Artifact
+ artifact=${groupAndArtifact/*:}
+
+ #Find version by format v[Number][Everything else before whitespace]
+ version=$(echo $currCommitMessage | grep -oP 'v[0-9]([^\s]+)')
+ version=${version/*v}
+
+ artifactAndVersion=$artifact$":"$version
+
+ #If we failed to find either the artifact or the version, return failure.
+ if [[ -z $artifact ]] || [[ -z $version ]]
+ then
+   return 1
```

Comment:
also contradicts line 31

Reply:
This code has been relocated to 'shared-deps-helper' getLibraryUpdate() and returns failure for the same reason as above (unable to find correct formatting), in which case in the main method (checkLibrary()) we should return success because we were unable to find a valid commit message, so the commit being checked doesn't have anything to do with updating dependencies.